### PR TITLE
fix for accidental / ledger line collisions

### PR DESCRIPTION
**Pull request by MarcSabatella**
Check for ledger lines on left notes within the loop and factor that in as well

Not meant to be merged as is, but to incorporate into @Marr11317 's work

### CODE REVIEW

File: libmscore/layout.cpp [hunk 1]
```diff
@@ -764,26 +764,30 @@ static qreal layoutAccidental(AcEl* me, AcEl* above, AcEl* below, qreal colOffse
       qreal lx = colOffset;
       Accidental* acc = me->note->accidental();
       qreal mag = acc->mag();
+      const qreal ledgerAdjust = -(acc->staff()->spatium(acc->tick()) * acc->score()->styleS(Sid::ledgerLineLength).val());
```
Comment (Marr11317) [inline libmscore/layout.cpp:767]: this will now be calculated by every single accidental. Could that be calculated only if needed?

Comment (MarcSabatella) [reply libmscore/layout.cpp:767]: Sure, with some effort.  But maybe not worth it.  Normally we don't need to layout out every single accidental in the whole score on every layout (unlike the case when this code was written).  We do far more expensive calculations regularly.

**Comment by MarcSabatella** [comment]
The vtest differences look good to me.  Note accidental-9 looks strange on purpose - it sets the accidental-to-note distance to 2 sp.

File: libmscore/layout.cpp (hunk 1 above)
Comment (MarcSabatella) [reply libmscore/layout.cpp:767]: One quick check would be, only do the calculation is the chord actually has ledger lines - that much should be simple to determine up front.

**Comment by MarcSabatella** [comment]
Since PR #6956 was merged before @Marr11317 had a chance to incorporate this code, now I *would* like this PR to be considered.  I have also improved it from the original version.

Here's the commit message that otherwise would have been the description:

The current algorithm for avoiding collision with ledger lines
only checks for a ledger line on the same note as the accidental,
and it does not account for "left notes" (notes on left of downsteam).

This commit improves the algorithm to check not just the line
of the accidental, but its top and bottom, and does the adjustment
if the accidental extends anywhere into ledger lines.
It also accounts for left notes.

**Review by Marr11317 — APPROVED** [review]
looks good! Now no `style()->get(Sid::...)` call is done if it's not needed.

**Comment by MarcSabatella** [comment]
I have rebased and pushed.  I'm expecting the same vtest differences (improvements) as before, maybe a few additional ones because I've since also enhanced the code to resolve more collisions.

**Comment by RobFog** [comment]
This addresses https://trello.com/c/Uj9B5F3m/50-ledger-lines-are-colliding-with-accidentals, correct?

**Pull request merged**